src/test/java/uk/gov/justice/digital/delius/CourtReportAPITest.java

```diff
@@ -0,0 +1,275 @@
+            .when()
+            .get("offenders/nomsNumber/NOMS1/courtReports/4")
+            .then()
+            .statusCode(200)
+            .extract()
+            .body()
+            .as(CourtReport.class);
+
+        assertThat(courtReport).isNotNull();
+        assertThat(courtReport.getCourtReportId()).isEqualTo(4);
+    }
+
+
+    @Test
+    public void missingOffenderRecordResultsInReportForOffenderByNomsNumberAndReportIdNotFound() {
+        when(offenderService.offenderIdOfNomsNumber("NOMS1")).thenReturn(Optional.empty());
+
+        given()
+                .header("Authorization", aValidToken())
+                .when()
+                .get("offenders/nomsNumber/NOMS1/courtReports/4")
+                .then()
+                .statusCode(404);
+    }
+
+    @Test
+    public void missingCourtReportRecordResultsInReportForOffenderByNomsNumberAndReportIdNotFound() {
+        when(courtReportService.courtReportFor(any(), any())).thenReturn(Optional.empty());
+
+        given()
+                .header("Authorization", aValidToken())
+                .when()
+                .get("offenders/nomsNumber/NOMS1/courtReports/4")
+                .then()
+                .statusCode(404);
+    }
+
+    @Test
+    public void canGetAllReportsForOffenderByOffenderId() {
+
+        CourtReport[] courtReports = given()
+            .header("Authorization", aValidToken())
+            .when()
+            .get("offenders/offenderId/1/courtReports")
+            .then()
+            .statusCode(200)
+            .extract()
+            .body()
+            .as(CourtReport[].class);
+
+        assertThat(courtReports).hasSize(3);
+    }
+
+    @Test
+    public void canGetSpecificReportForOffenderByOffenderIdAndReportId() {
+
+        CourtReport courtReport = given()
+            .header("Authorization", aValidToken())
+            .when()
+            .get("offenders/offenderId/1/courtReports/4")
+            .then()
+            .statusCode(200)
+            .extract()
+            .body()
+            .as(CourtReport.class);
+
+        assertThat(courtReport).isNotNull();
+        assertThat(courtReport.getCourtReportId()).isEqualTo(4);
+    }
+
+    @Test
+    public void missingCourtReportRecordResultsInReportForOffenderByOffenderIdAndReportIdNotFound() {
+        when(courtReportService.courtReportFor(any(), any())).thenReturn(Optional.empty());
+
+        given()
+                .header("Authorization", aValidToken())
+                .when()
+                .get("offenders/offenderId/1/courtReports/4")
+                .then()
+                .statusCode(404);
+    }
+
+
+
+    private String aValidToken() {
+        return aValidTokenFor(UUID.randomUUID().toString());
+    }
+
+    private String aValidTokenFor(String distinguishedName) {
+        return "Bearer " + jwt.buildToken(UserData.builder()
+                .distinguishedName(distinguishedName)
+                .uid("bobby.davro").build());
+    }
```

Comment:
Could also test the failed auth scenarios. IIRC we do this in the other controller tests

Reply:
Yep, added scenarios for missing token